go/arrow/datatype_fixedwidth.go

```diff
@@ -220,17 +220,17 @@ func (t Time64) ToTime(unit TimeUnit) time.Time {
 }
 
 const (
-	Nanosecond TimeUnit = iota
-	Microsecond
+	Second TimeUnit = iota
 	Millisecond
-	Second
+	Microsecond
```

Comment:
does this reordering have any semantic impacts?

Reply:
It did, which is why there are the corresponding re-orderings below it in the `Multiplier` and `String` functions. There should be no other semantic impacts other than that though since nothing else should be depending on the actual numeric value of the enum.